diff --git a/fem/symbolicintegrator.cpp b/fem/symbolicintegrator.cpp
index 8befee4..50cd78b 100644
--- a/fem/symbolicintegrator.cpp
+++ b/fem/symbolicintegrator.cpp
@@ -39,6 +39,11 @@ namespace ngfem
       return Array<int> ({dim/blockdim, blockdim});
   }
   
+  string ProxyFunction ::
+  GetString(FlatArray<int> inputs, int index) const
+  {
+    return evaluator->Name() + string(testfunction? "(testfunction)" : "(trialfunction)") + "\n";
+  }
   
   void ProxyFunction ::
   Evaluate (const BaseMappedIntegrationPoint & mip,
diff --git a/fem/symbolicintegrator.hpp b/fem/symbolicintegrator.hpp
index b24d8cd..463561a 100644
--- a/fem/symbolicintegrator.hpp
+++ b/fem/symbolicintegrator.hpp
@@ -47,6 +47,8 @@ public:
   virtual Array<int> Dimensions() const;
   virtual bool IsComplex () const { return is_complex; } 
   bool IsOther() const { return is_other; }
+
+  virtual string GetString(FlatArray<int> inputs, int index) const;
   
   const shared_ptr<DifferentialOperator> & Evaluator() const { return evaluator; }
   const shared_ptr<DifferentialOperator> & DerivEvaluator() const { return deriv_evaluator; }
