diff --git a/adm/cmake/occt_defs_flags.cmake b/adm/cmake/occt_defs_flags.cmake
index 5bd3313d3c..a31cdb25f5 100644
--- a/adm/cmake/occt_defs_flags.cmake
+++ b/adm/cmake/occt_defs_flags.cmake
@@ -42,7 +42,6 @@ else()
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
     set (CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -Wno-deprecated-declarations")
   endif()
-  add_definitions(-DOCC_CONVERT_SIGNALS)
 endif()
 
 # enable structured exceptions for MSVC
diff --git a/src/Standard/Standard_ErrorHandler.cxx b/src/Standard/Standard_ErrorHandler.cxx
index 5ec6e4865a..01c1671cd9 100644
--- a/src/Standard/Standard_ErrorHandler.cxx
+++ b/src/Standard/Standard_ErrorHandler.cxx
@@ -159,16 +159,10 @@ void Standard_ErrorHandler::Abort(const Handle(Standard_Failure)& theError)
   Standard_ErrorHandler* anActive = FindHandler(Standard_HandlerVoid, Standard_True);
 
   //==== Check if can do the "longjmp" =======================================
-  if (anActive == NULL)
-  {
     std::cerr << "*** Abort *** an exception was raised, but no catch was found." << std::endl;
     if (!theError.IsNull())
       std::cerr << "\t... The exception is:" << theError->GetMessageString() << std::endl;
     exit(1);
-  }
-
-  anActive->myStatus = Standard_HandlerJumped;
-  longjmp(anActive->myLabel, Standard_True);
 }
 
 //============================================================================
diff --git a/src/Standard/Standard_ErrorHandler.hxx b/src/Standard/Standard_ErrorHandler.hxx
index 1eec747533..7c7c591e7c 100644
--- a/src/Standard/Standard_ErrorHandler.hxx
+++ b/src/Standard/Standard_ErrorHandler.hxx
@@ -45,29 +45,8 @@
 //! signal handler, this OCC_CONVERT_SIGNALS is not needed. Note however that
 //! this requires that compiler option /EHa is used.
 
-#if defined(OCC_CONVERT_SIGNALS)
-
-  // Exceptions are raied as usual, signal cause jumps in the nearest
-  // OCC_CATCH_SIGNALS and then thrown as exceptions.
-  #define OCC_CATCH_SIGNALS                                                                        \
-    Standard_ErrorHandler _aHandler;                                                               \
-    if (setjmp(_aHandler.Label()))                                                                 \
-    {                                                                                              \
-      _aHandler.Catches(STANDARD_TYPE(Standard_Failure));                                          \
-      _aHandler.Error()->Reraise();                                                                \
-    }
-
-  // Suppress GCC warning "variable ...  might be clobbered by 'longjmp' or 'vfork'"
-  #if defined(__GNUC__) && !defined(__INTEL_COMPILER) && !defined(__clang__)
-    #pragma GCC diagnostic ignored "-Wclobbered"
-  #endif
-
-#else
-
-  // Normal Exceptions (for example WNT with MSVC and option /GHa)
-  #define OCC_CATCH_SIGNALS
-
-#endif
+// Normal Exceptions (for example WNT with MSVC and option /GHa)
+#define OCC_CATCH_SIGNALS
 
 class Standard_Failure;
 
diff --git a/src/Standard/Standard_JmpBuf.hxx b/src/Standard/Standard_JmpBuf.hxx
index 44a9406c29..4e2752c305 100644
--- a/src/Standard/Standard_JmpBuf.hxx
+++ b/src/Standard/Standard_JmpBuf.hxx
@@ -14,15 +14,14 @@
 #ifndef _Standard_JmpBuf_HeaderFile
 #define _Standard_JmpBuf_HeaderFile
 
-#include <setjmp.h>
-
+#include <cstdint>
 typedef
 #ifdef SOLARIS
   sigjmp_buf
 #elif defined(IRIX)
   sigjmp_buf
 #else
-  jmp_buf
+  intptr_t
 #endif
     Standard_JmpBuf;
 
